src/index/precision_locking_index/index.hpp

```diff
@@ -37,6 +37,16 @@ class HashTableWithPrecisionLockingIndex {
 
   T* Get(const std::string_view key) { return point_index_.Get(key); }
 
+  bool IsEntryNull(const std::string_view key) {
+    return point_index_.Get(key) == nullptr;
+  }
+
+  bool IsEntryEmpty(std::string_view key) {
+    T* entry = point_index_.Get(key);
+    if (entry == nullptr || entry->initialized) return false;
+    return range_index_.Insert(key);
+  }
```

Comment:
Having side effects (insert) in a boolean function named `IsEntryEmpty()` is extremely poor design.
Such boolean functions named `Is...()` should be const, as shown below.

```c++
bool IsEntryEmpty(...) const {...

}
```
Maybe you should do the insert outside this function.

Reply:
Instead, remove `IsEntryNull()` and `IsEntryEmpty()`.